core/tests/unittest.py

```diff
@@ -236,38 +236,48 @@ def wasSuccessful(self):
 
 def run_class(c, test_result):
     o = c()
+    set_up_class = getattr(o, "setUpClass", lambda: None)
+    tear_down_class = getattr(o, "tearDownClass", lambda: None)
     set_up = getattr(o, "setUp", lambda: None)
     tear_down = getattr(o, "tearDown", lambda: None)
     print("class", c.__qualname__)
-    for name in dir(o):
-        if name.startswith("test"):
-            print(" ", name, end=" ...")
-            m = getattr(o, name)
-            try:
-                try:
-                    set_up()
-                    test_result.testsRun += 1
-                    retval = m()
-                    if isinstance(retval, generator_type):
-                        raise RuntimeError(
-                            f"{name} must not be a generator (it is async, uses yield or await)."
-                        )
-                    elif retval is not None:
-                        raise RuntimeError(f"{name} should not return a result.")
-                finally:
-                    tear_down()
-                print(f"{OK_COLOR} ok{DEFAULT_COLOR}")
-            except SkipTest as e:
-                print(" skipped:", e.args[0])
-                test_result.skippedNum += 1
-            except AssertionError as e:
-                print(f"{ERROR_COLOR} failed{DEFAULT_COLOR}")
-                sys.print_exception(e)
-                test_result.failuresNum += 1
-            except BaseException as e:
-                print(f"{ERROR_COLOR} errored:{DEFAULT_COLOR}", e)
-                sys.print_exception(e)
-                test_result.errorsNum += 1
+    try:
+        set_up_class()
+        for name in dir(o):
+            if name.startswith("test"):
+                run_test_method(o, name, set_up, tear_down, test_result)
+    finally:
+        tear_down_class()
+
+
+def run_test_method(o, name, set_up, tear_down, test_result):
```

Comment:
Personally, I would prefer a function with less parameters. However:
- it is not possible to pass `m` instead of `o` and `name`, as getting `name=m.__name__` will fail for test functions with decorators (e.g.` test_apps.common.keychain.py::test_verify_path`)
- getting `setUp` and `tearDown` for each test in class is redundant
- it would be possible to encapsulate the arguments into a single `test_args` object, but imo it might be an overkill for the use case
- inlining the `run_test` function will make the code less readable because of the complex nesting

With those points in mind, I think that the 5 parameters might be an ok solution.